### PR TITLE
Upgrade caffiene dependency in pom.xml

### PR DESCRIPTION
**Pull request by ConcurrencyPractitioner**
### Motivation

We want to get a new version of caffeine. 

### Modifications
Changed pom.xml's caffeine version to 2.6.2.

### Result

Upgraded caffeine change.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 1]
```diff
@@ -427,7 +427,7 @@ public void close() throws IOException {
     public void unloadNamespaceBundlesGracefully() {
         try {
             // make broker-node unavailable from the cluster
-            if (pulsar.getLoadManager() != null) {
+            if (pulsar.getLoadManager() != null && pulsar.getLoadManager().get() != null) {
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java]: this is an unrelated change. can you remove it?

Comment (ConcurrencyPractitioner) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java]: Yeah, I removed it. :)

File: pom.xml [hunk 2]
```diff
@@ -667,7 +667,7 @@ flexible messaging model and an intuitive client API.</description>
       <dependency>
         <groupId>com.github.ben-manes.caffeine</groupId>
         <artifactId>caffeine</artifactId>
-        <version>2.3.3</version>
+        <version>2.6.2</version>
```
Comment (merlimat) [inline pom.xml]: We also need to update the license file with the list of binaries: `distribution/server/src/assemble/LICENSE.bin.txt`

Comment (merlimat) [reply pom.xml]: Yes, the validation check for the Java job otherwise is failing :)

**Comment by sijie** [comment]
run java8 tests
